src/Chapter4.hs

```diff
@@ -672,7 +707,24 @@ Specifically,
    subtree of a tree
  ❃ Implement the function to convert Tree to list
 -}
+data Tree a
+  = Leaf
+  | Node (Tree a) a (Tree a)
+  deriving (Show, Eq)
+
+instance Functor Tree where
+  fmap f (Node left a right) = Node (fmap f left) (f a) (fmap f right)
+  fmap _ _ = Leaf
```

Comment:
I would pattern match on the `Leaf` constructor here explicitly, so in case you decide to change the `Tree` type by adding one more constructor to it, you will get a compiler warning to update the `fmap` definition and handle new constructor